patches/server/0663-More-World-API.patch

```diff
@@ -64,8 +64,31 @@ index 4717bd703d3a8bb87c71d14eb851db09c264046e..b825673fdd7c8c8dd9e05fe6b317006a
 +    public Collection<org.bukkit.Material> getInfiniburn() {
 +        return com.google.common.collect.Sets.newHashSet(com.google.common.collect.Iterators.transform(getHandle().dimensionType().infiniburn().getValues().iterator(), CraftMagicNumbers::getMaterial));
 +    }
++
++    @Override
++    public void sendGameEvent(Entity sourceEntity, org.bukkit.GameEvent gameEvent, Vector position) {
++        getHandle().gameEvent(((CraftEntity) sourceEntity).getHandle(), net.minecraft.core.Registry.GAME_EVENT.get(org.bukkit.craftbukkit.util.CraftNamespacedKey.toMinecraft(gameEvent.getKey())), org.bukkit.craftbukkit.util.CraftVector.toBlockPos(position));
```

Comment:
Passing a null sourceEntity will result into a ClassCastException. 
Null cannot be cast to CraftEntity

Reply:
null can be casted to anything iirc?

Reply:
Yeah, null can be casted to anything, but it will cause a NullPointerException since its null. Will fix.

Reply:
Yeah, deleted the NPE remark there because I thought CCE would be thrown first. 🙄